CMSgov/mdct-mcr
# configure form handling and field validation

## Description
This is the final PR to build out the functionality for [284](https://qmacbis.atlassian.net/browse/MDCT-284). 

Things that changed:
- Tweaked `Alert` styles.
- Moved the form component out of `Admin` and into it's own `AdminBannerForm` component. Added `yup` object schema validation and got it hooked up to the fields.
- Modified `TextField` to wire it up for field validation.
- Renamed `DataField` to `DateField` and got that component wired up for field validation. Note: I'm not sold on the way I'm doing this here but there was no documentation about how to handle this kind of weirdly nested field, so this is what we have for now. I have an idea or two that I'd like to try in the next sprint to make this even better.
- Added a couple constants to the `time` utilities and a handful of related interfaces to the `types` utilities.

_Note: There are no tests yet but there will be. We're gonna be working hard on testing next sprint. Today we're just trying to get this out there._

Related ticket: [MDCT-284](https://qmacbis.atlassian.net/browse/MDCT-284)

### How to test
1. Make sure you change your `.env` files to match the updated ones in the readme.
2. Quick `yarn install` to pick up the new package.
3. Run and navigate to /admin.
4. Create a new banner. The form is all wired up with validation and such.

### Changed Dependencies
Added [yup](https://github.com/jquense/yup) for field/form validation.

## Code author checklist
- [] I have added [thorough](https://qmacbis.atlassian.net/wiki/spaces/CM/pages/2914025525/Test+Suite+and+Testing+Research) tests

## Reviewer checklist (two different people)
